build/commands/lib/config.js

```diff
@@ -1221,6 +1221,12 @@ Object.defineProperty(Config.prototype, 'defaultOptions', {
       ['build'],
       ['third_party', 'depot_tools'],
     ]
+    if (this.getTargetOS() === 'mac' && process.platform !== 'darwin') {
```

Comment:
The directories that get added to the PYTHONPATH and normal PATH here overwrite and provide built-ins that are normally only available on macOS. If we were to add the directories unconditionally, then the overwritten implementations would take effect on macOS, and because it is not written for that OS, not work.

So in short, if we were to add the directories unconditionally, then the build on macOS would break. The `if` guard prevents this.